### ISSUE TITLE
Query Solr did not work well

### ISSUE THREAD

**Issue by hendisantika** [takezoe/solr-scala-client#34]
I have tried your example for regular client and Async but it did not work for query.
I have created a repo so that so can try it.
`https://github.com/hendisantika/Coba-Solr`

Here are my notes : 
1. Coba.scala --> It add to solr repo but the query result is not working
2. AsyncSolrClientSample.scala --> Can not add and query.
3. SolrClientSample.scala --> Can not add and query.

Please give me an advice and solution.

Thanks

**Comment by takezoe** [comment]
> Coba.scala --> It add to solr repo but the query result is not working

If following query works, schema is invalid. Has `name` field been tokenized correct?

```scala
val result = client.query("id: %id%")
  .fields("id", "manu", "name")
  .sortBy("id", Order.asc)
  .getResultAsMap(Map("id" -> "001"))
```

> AsyncSolrClientSample.scala --> Can not add and query.

I tested and worked in my environment. It's strange...

> SolrClientSample.scala --> Can not add and query.

Remove following code which provides a mock client. If this code is available, all methods of client don't access to the real Solr server.

```scala
  implicit val solr = SolrClientFactory.dummy { request =>
    println(request.getMethod)
    println(request.getPath)
    println(request.getParams)
  }
```

**Comment by takezoe** [comment]
Let me know details of your situation. There are too few clues to find a cause.

- How did you establish your solr server and collection?
- Did you try the id query I shown in `Coba.scala`? How was the result?
- Does "query result is not working" mean an empty result? or an exception?

**Comment by hendisantika** [comment]
1. I follow your example. It is exactly same with your code because I copied from yours.
2. Yes, I did.
3. Empty result. Nothing print on my terminal

**Comment by hendisantika** [comment]
1. I installed via tutorial `https://www.howtoforge.com/tutorial/how-to-install-and-configure-solr-on-ubuntu-1604/`. I run via browser to open my solr repo. I created my repo via the above too.
2. It is printed on my IDEA Console 'registered' but then the stacktrace come.
I dont think so.

**Comment by takezoe** [comment]
Result depends on environment. In my environment:

- **Coba.scala** -> Both register and query succeed.
- **AsyncSolrClientSample.scala** -> Register fails silently and query throws `org.apache.solr.common.SolrException: parsing error`. However only query succeeds by removing last `/` in a url as `"http://localhost:8983/solr/gettingstarted"`.
- **SolrClientSample.scala** -> Not worked raw code because there is `implicit val solr = SolrClientFactory.dummy { ... }`. After removing this part both register and query succeed.

From these results, I can say:

- Async registration has a bug
- Url handling about last / in the async client should be improved

I will raise new issues about them. But I couldn't find other problem. In particular, the sync client has worked with no problem.

I guess `name` field is not tokenized in your environment because your `gettingstarted` collection was created with `data_driven_schema_configs` option. Therefore maybe only exact match search is available for this field. However this is just my guess.

This is all I can do. Unfortunately, maybe I can't help you any more.

**Comment by hendisantika** [comment]
Ooo.. I see that.

I want to use yout Async method for my test.

At least I know the issue from your explainations.

By The way.

Arigatou Gozaimasu.
:+1:

**Comment by takezoe** [comment]
Fix async client example in 0a6943683991b22531f12723234b8f10d00a1c90. Querying had required waiting registration. I confirmed that async client work well in this code.

**Comment by hendisantika** [comment]
OK. I will retry.
Thanks to you @takezoe
